src/impl_methods.rs

```diff
@@ -2444,6 +2444,46 @@ where
         }
     }
 
+    /// Remove the `index`th elements along `axis` and shift down elements from higher indexes.
+    ///
+    /// Decreases the length of `axis` by one.
+    ///
+    /// ***Panics** if `axis` or `index` is out of bounds.
+    pub fn shift_remove_index(&mut self, axis: Axis, index: usize)
```

Comment:
```suggestion
    pub fn remove_index(&mut self, axis: Axis, index: usize)
```

IMO, `shift_` doesn't need to be part of the name.